### ISSUE TITLE
Incorrect operation of exclusion lists

### ISSUE THREAD

**Issue by Eksol** [AdguardTeam/AdGuardVPNForWindows#770]
### Please answer the following questions for yourself before submitting an issue.

- [X] I checked the [knowledge base](https://adguard-vpn.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard VPN version

2.4.0

### Environment

```markdown
- OS: Windows 10
- Other network-level software: adguard
```

### Which DNS server do you use?

Default

### Use Wintun

- [X] yes, I do

### Issue Details

Steps to reproduce:
1. Open rutracker.org by including the domain in the exception lists

### Expected Behavior

The site should open

### Actual Behavior

The site does not open

### Screenshots

<details><summary>Screenshot 1</summary>

</details>

**Comment by Eksol** [comment]
![Скриншот 08 11 24_20 24 21](https://github.com/user-attachments/assets/dd133a5b-7a08-44c0-962e-d378e82037ed)
![Скриншот 08 11 24_20 20 52](https://github.com/user-attachments/assets/2a56600a-db9d-4667-9342-1e1f09f9c93b)

**Comment by Eksol** [comment]
![Скриншот 08 11 24_20 29 49](https://github.com/user-attachments/assets/30964906-ecb5-4e92-be32-4db52e86a21e)
Is this program written in python? It hangs, slows down, doesn't work!

**Comment by shamarin** [comment]
Rutracker is blocked in Russia.

**Comment by AlexandrPkhm** [comment]
@Eksol Hello! 

First of all, could you please update to the latest version of AdGuard VPN for Windows which is [2.4.1](https://github.com/AdguardTeam/AdGuardVPNForWindows/releases/tag/v2.4.1) and check if the issues with exclusions and crashes persist there? 

Regarding the rutracker.org exclusion, it appears that you have added the correct rutracker.org domain, so could you please clarify the following What mode are you using for app exclusions, is it also set to selective or is it general?

**Comment by Eksol** [comment]
> @Eksol Hello!
> 
> First of all, could you please update to the latest version of AdGuard VPN for Windows which is [2.4.1](https://github.com/AdguardTeam/AdGuardVPNForWindows/releases/tag/v2.4.1) and check if the issues with exclusions and crashes persist there?
> 
> Regarding the rutracker.org exclusion, it appears that you have added the correct rutracker.org domain, so could you please clarify the following What mode are you using for app exclusions, is it also set to selective or is it general?

selective

**Comment by Eksol** [comment]
I installed the update. It didn't help. Only switching to the general mode when all traffic is sent to the tunnel helps to solve the problem.

**Comment by AlexandrPkhm** [comment]
@Eksol 

> selective

If you are using _Selective_ mode for the _App exclusions_ please make sure that your browser is added to the list of the App exclusions.

**Comment by Eksol** [comment]
> @Eksol
> 
> > selective
> 
> If you are using _Selective_ mode for the _App exclusions_ please make sure that your browser is added to the list of the App exclusions.

Yes, the browser is added. YouTube traffic goes through the tunnel. But rutracker does not open.

**Comment by AlexandrPkhm** [comment]
@Eksol Could you please clarify if this happens on one particular location or on every location? 

Also, please record the application logs for us. We need them in order to diagnose this issue.
Here's what we need you to do:

1. Go to *Settings -> App settings -> Logging level -> Record everything*.
2. Reproduce the problem, then remember the exact time it happened.
3. Open *Support tab -> Export logs and system info*.
4. Send the archive to [qa@adguard.com](mailto:qa@adguard.com):
		- include `[windows]` keyword and `ISSUE_NUMBER` in the subject of your email
		- specify the exact time when the issue occurred

**Comment by Eksol** [comment]
IP(rutracker) blocked by adguard for chrome

**Comment by Eksol** [comment]
> please record the application logs for us

Done...

**Comment by Eksol** [comment]
The problem is localized in the Google Chrome browser. After reinstallation, the RuTracker site only worked halfway. I see a block on the RuTracker.ck domain, but this site, which is included in the exception list, is not processed correctly.

**Comment by Eksol** [comment]
![Скриншот 14 11 24_01 37 24](https://github.com/user-attachments/assets/cd647191-c1a4-43ec-862d-7a93e1ac9607)

**Comment by AChekanoff** [comment]
I have the same problem.

**Comment by CattusAter** [comment]
Та же проблема на Андроиде, но:
Режим интеграции с блокировщиком
Выключил блокировщик, заработало. Включил – сломалось. Выключил только DNS-защиту – заработало. На всякий случай выключил интеграцию – работает.
Итого, проблема только в режиме интеграции при включённой DNS защите = проблема не в VPN, а в защите DNS.
На Windows проверю позже, думаю, там то же самое.

**Comment by CattusAter** [comment]
Поймал. Если в пользовательский фильтр DNS добавить разрешение @@||tns-counter.ru^ (присутствует блок в стандартном фильтре DNS), то рутрекер открывается в выборочном режиме в интеграции. Но блокировать эту дрянь вообще-то надо, так что не знаю, что дальше делать с этой информацией.

**Comment by CattusAter** [comment]
Это смешно, но я убрал tns-counter.ru из исключений блокировщика и добавил в VPN. И рутрекер заработал в выборочном режиме. При этом tns-counter.ru не разолвится, то есть блокировка отрабатывает.

**Comment by CattusAter** [comment]
Нет, похоже, запись просто оставалась в кэше. Снова не работает без исключения в блокировке.

**Comment by CattusAter** [comment]
Под windows не воспроизводится. Без приключений открылся рутрекер при добавлении в исключение VPN доменов rutracker.org и rutracker.cc. Последние Nightly оба, блокировщик и VPN.

**Comment by CattusAter** [comment]
Сейчас всё работает, рутрекер открывается в режиме интеграции без добавления исключений в DNS-фильтрацию.
AdGuard v4.9 Nightly 2
AdGuard VPN v2.12 Nightly 9

**Comment by shamarin** [comment]
> Сейчас всё работает, рутрекер открывается в режиме интеграции без добавления исключений в DNS-фильтрацию. AdGuard v4.9 Nightly 2 AdGuard VPN v2.12 Nightly 9

Так вы приводите версии для мобильных устройств. Там совершенно другое. Здесь же приложение для Windows.

**Comment by Vladislavik** [comment]
При включенном wintun исключённый YouTube идёт через VPN, с wintun исключения не работают. Windows 11, VPN 2.5.1 beta2

**Comment by shamarin** [comment]
> При включенном wintun исключённый YouTube идёт через VPN, с wintun исключения не работают. Windows 11, VPN 2.5.1 beta2

Так и торренты не понятно как работают. Не в исключениях торрент клиент, а трафик идет не через VPN.

**Comment by CattusAter** [comment]
> > Сейчас всё работает, рутрекер открывается в режиме интеграции без добавления исключений в DNS-фильтрацию. AdGuard v4.9 Nightly 2 AdGuard VPN v2.12 Nightly 9
> 
> Так вы приводите версии для мобильных устройств. Там совершенно другое. Здесь же приложение для Windows.

Мне в чате в телеграме дали ссылку на эту задачу. Я сразу написал, что воспроизводится на Android, а на Windows не воспроизводится. Сейчас не воспроизводится ни там, ни там.

**Comment by CattusAter** [comment]
> > При включенном wintun исключённый YouTube идёт через VPN, с wintun исключения не работают. Windows 11, VPN 2.5.1 beta2
> 
> Так и торренты не понятно как работают. Не в исключениях торрент клиент, а трафик идет не через VPN.

Торренты работают P2P, так что либо выключайте избирательный режим, либо будет не через VPN. Потому что нет режима "все адреса, но не все приложения".

**Comment by shamarin** [comment]
P2P к сожалению в любом режиме сейчас напрямую трафик идет, а не через VPN как должно быть. Создана задача по этому поводу и разработчики разбираются уже с этим https://github.com/AdguardTeam/VpnLibs/issues/61

**Comment by adbuker** [comment]
@shamarin , @CattusAter, @Vladislavik, @AChekanoff, @Eksol, could you please check whether the issue is repro with the latest nightly build

**Comment by shamarin** [comment]
> [@shamarin](https://github.com/shamarin) , [@CattusAter](https://github.com/CattusAter), [@Vladislavik](https://github.com/Vladislavik), [@AChekanoff](https://github.com/AChekanoff), [@Eksol](https://github.com/Eksol), could you please check whether the issue is repro with the latest nightly build

У меня только с torrent странно работает. Если ставить в исключения торрент клиент, то если смотреть в диспетчере задач, то он показывает, что либо весь трафик идет через VPN или часть его. Хотя при исключении по идее весь трафик торрент клиента должен идти напрямую.

**Comment by IaroslavKhvorostenko** [comment]
No issues detected with exclusions on the part of AdGuard VPN. Closing this issue.
